pkg/detector/detector.go

```diff
@@ -1026,19 +1029,61 @@ func (d *ResourceDetector) ReconcileResourceBinding(key util.QueueKey) error {
 	}
 
 	klog.Infof("Reconciling resource binding(%s/%s)", binding.Namespace, binding.Name)
-	switch binding.Spec.Resource.Kind {
-	case util.DeploymentKind:
-		return d.AggregateDeploymentStatus(binding.Spec.Resource, binding.Status.AggregatedStatus)
-	case util.ServiceKind:
-		return d.AggregateServiceStatus(binding.Spec.Resource, binding.Status.AggregatedStatus)
-	case util.IngressKind:
-		return d.AggregateIngressStatus(binding.Spec.Resource, binding.Status.AggregatedStatus)
-	case util.JobKind:
-		return d.AggregateJobStatus(binding.Spec.Resource, binding.Status.AggregatedStatus, binding.Spec.Clusters)
-	default:
-		// Unsupported resource type.
+	resource := binding.Spec.Resource
+	gvr, err := restmapper.GetGroupVersionResource(
+		d.RESTMapper, schema.FromAPIVersionAndKind(resource.APIVersion, resource.Kind),
+	)
+	if err != nil {
+		klog.Errorf("Failed to get GVR from GVK %s %s. Error: %v", resource.APIVersion, resource.Kind, err)
+		return err
+	}
+	obj, err := d.DynamicClient.Resource(gvr).Namespace(resource.Namespace).Get(context.TODO(), resource.Name, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get resource(%s/%s/%s), Error: %v", resource.Kind, resource.Namespace, resource.Name, err)
+		return err
+	}
+
+	if !d.ResourceInterpreter.HookEnabled(obj, configv1alpha1.InterpreterOperationAggregateStatus) {
 		return nil
 	}
+	newObj, err := d.ResourceInterpreter.AggregateStatus(obj, binding.Status.AggregatedStatus)
+	if err != nil {
+		klog.Errorf("AggregateStatus for resource(%s/%s/%s) failed: %v", resource.Kind, resource.Namespace, resource.Name, err)
+		return err
+	}
+	newStatus, _, err := unstructured.NestedFieldNoCopy(newObj.Object, "status")
+	if err != nil {
+		return err
+	}
+	oldStatus, _, err := unstructured.NestedFieldNoCopy(obj.Object, "status")
+	if err != nil {
+		return err
+	}
```

Comment:
We don't need to parse the `.status`, we can just compare the `obj` and `newObj`, since the `Interpreter` won't change anything but `.status`, so if `obj` != `newObj` means the status has been `aggregated`, then we do `update` operation.

In addition, we seem can't assume there is a `.status` field. See [InterpretStatus](https://github.com/karmada-io/karmada/blob/8ceb9df2b9672d3e50b0d52d6b4e7d8c000eff81/pkg/apis/config/v1alpha1/resourceinterpreterwebhook_types.go#L85-L87).

Reply:
Something like this:
```golang
	if reflect.DeepEqual(obj, newObj) {
		klog.V(3).Infof("ignore update resource(%s/%s/%s) status as up to date", resource.Kind, resource.Namespace, resource.Name)
		return nil
	}
```

Maybe we can't use `retry.RetryOnConflict` for this case,

Reply:
> Maybe we can't use retry.RetryOnConflict for this case

What is this for? I can compare two objects directly, but it doesn't seem to affect the use of `RetryOnConflict`, is it because the newObj may also not have status?

Reply:
> but it doesn't seem to affect the use of RetryOnConflict, is it because the newObj may also not have status?

As after we `get` a new copy of the object, we have to set the status, do you mean to call the `webhook` again in `retry`?

Reply:
Nope, Is there a problem with the current retry method?

Reply:
The thing is 
```
if err = unstructured.SetNestedField(obj.Object, newStatus, "status"); err != nil {
			return err
		}
```

Personally, I'm hesitant to assume this is a `.status` filed(Maybe no problem for 99.9% cases).

Reply:
I am not so familiar with `SetNestedField`, but by looking at the source code, if the original object does not have a status field, it should not report an error when setting for it, it will only report an error if the value is not compliant.

The workload object initially had no status field, but `SetNestedField` added it, as expected.

Reply:
I mean we can't assume all resource's status represent by `.status`, that might be `.status-xxx` or something else. That's why we design `InterpretStatus` operation for.

Reply:
I get your point now, it seems we can't use retry in this case, let's return the error directly to trigger its resync.